setup.py

```diff
@@ -29,9 +29,35 @@ def read(fname):
 
 class CustomInstallCommand(install):
     '''a customized install class in pip module'''
+    user_options = install.user_options + [
+        ('sdk-only', None, '<add --sdk-only if you only want to install nni sdk only')
+    ]
+
+    def initialize_options(self):
+        install.initialize_options(self)
+        self.sdk_only = None
+
+    def install_requires(self):
+        self.install_requires_list = [
```

Comment:
- Move install_requires to here because pip install with install-option flag will apply to all install_requires if it is in the setup.
- It will failed because other packages' setup.py have no such flag

referring to https://github.com/pypa/pip/issues/1883